### PR TITLE
Update with rapids cmake new features

### PR DESCRIPTION
**Pull request by robertmaynard**
This combines some general CMake style cleanup and brings new rapids-cmake features to RAFT including:

- Usage of `rapids_cmake_install_lib_dir` to make sure we install raft correctly on non-debain based distro's ( lib64 ), while also handling conda installation requirements ( always lib no matter the distro )
- Usage of `rapids_cpm` pre-configured pacakges
- Removal of early termination before `rapids_cpm_find` since a better solution now exists ( https://github.com/rapidsai/rapids-cmake/issues/49 )

### CODE REVIEW

File: cpp/cmake/thirdparty/get_rmm.cmake [hunk 1]
```diff
@@ -16,9 +16,11 @@
...
 function(find_and_configure_rmm VERSION)
 
-    if(TARGET rmm::rmm)
-        return()
-    endif()
+    # No way to override the version rapids-cmake pulls
+    # so we need to manually checkout when needing a diverged
+    # version
+    # include(${rapids-cmake-dir}/cpm/rmm.cmake)
+    # rapids_cpm_rmm()
```
Comment (dantegd) [inline cpp/cmake/thirdparty/get_rmm.cmake:23]: We can remove the pinned rmm now (I'm doing it in #322 as well to run CI correctly )

**Review by dantegd — CHANGES_REQUESTED** [review]
Changes look great, just a comment about the RMM pin

File: cpp/cmake/thirdparty/get_rmm.cmake (hunk 1 above)
Comment (robertmaynard) [reply cpp/cmake/thirdparty/get_rmm.cmake]: done

**Review by dantegd — APPROVED** [review]

**Comment by dantegd** [comment]
@gpucibot merge

**Pull request merged**